### PR TITLE
Libraries: Form Validation

### PR DESCRIPTION
**Pull request by Garfield-fr**
* NEW: Library Form Validation

How to test:
Run the scripts `bootstrap` and `setup`
Check form validation on edit.
Fields Required:
- Name
- Code (test unique valuer)
- Time Range (start_time > end_time)
- Date Exception: title, Time period, etc…

Co-authored-by: Peter Weber <peter.weber@rero.ch>
Signed-off-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

### CODE REVIEW

**Comment by NicolasLabat** [comment]
(perhaps just for info) Opening hours : a control is done when end time is less than start time. However, when adding a period, there is no control between periods. It is for example possible to submit : 
08:00 to 12:00
11:00 to 17:00

**Comment by NicolasLabat** [comment]
Opening hours : it is possible to submit a non-correct format.
Example : 08:00 13:30pm

**Comment by NicolasLabat** [comment]
(Enhancement?) Date exceptions : when using the "repeat" action, it could be interesting to display the end date. For example : if the library is closed on Sunday starting the 02.12 for 16 weeks, it is displayed "weekly=16" but we have no information about the end date precisely.

File: ui/admin/src/app/libraries/libraries.service.ts [hunk 1]
```diff
@@ -36,6 +45,25 @@ export class LibrariesService {
...
   }
 
+  checkIfCodeAlreadyTaken(code: string): Observable<Boolean> {
+    return this.client.get<any>(
+        this.librariesUrl + '/?q=code:' + code +
+        '&libraries.code:' + this.loggedUser.organisation_pid +
+        '&size=0'
+      ).pipe(map(response => {
+        if (this.currentLibrary === null) {
+          return response.hits.total >= 1;
```
[the remark this answers is not in the document]
Comment (Garfield-fr) [reply ui/admin/src/app/libraries/libraries.service.ts]: Changed

File: ui/admin/src/app/libraries/library-form.service.ts [hunk 2]
```diff
@@ -24,23 +35,57 @@ export class LibraryFormService {
...
+      opening_hours: this.fb.array(this.createOpeningHours())
     });
   }
 
-  buildOpeningHours(): FormGroup {
+  createOpeningHours() {
+    const days = ['monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday'];
+    const openings = [];
+    let step = 0;
+    for (step = 0; step < 7; step++) {
```
[the remark this answers is not in the document]
Comment (Garfield-fr) [reply ui/admin/src/app/libraries/library-form.service.ts]: Changed

**Pull request merged**